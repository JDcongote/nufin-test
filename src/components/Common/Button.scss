@import '../../styles/variables/index';
@import '../../styles/tools/mixins';

.button {
  @include button-unstyled;
  z-index: var(--z-index--level-1);
  border-radius: var(--border--radius--small);

  &-primary {
    padding: 0.5em;
    background-color: var(--color--button-primary);
    font-weight: bold;
    color: var(--color--secondary);
  }
  &.bottom {
    position: fixed;
    bottom: 0;
    margin: 1em;
    &-right {
      right: 0;
    }
    &-left {
      left: 0;
    }
  }
}
