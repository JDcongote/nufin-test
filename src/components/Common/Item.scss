@import '../../styles/variables/index';
@import '../../styles/tools/mixins';

.item {
  position: relative;
  width: 90vw;
  max-width: 100%;
  padding: 8px;
  margin: 20px auto;
  background-color: var(--color--background-card);
  z-index: var(--z-index--level-1);
  overflow: hidden;
  min-height: 120px;
  border-radius: var(--border--radius--medium);
  -webkit-box-shadow: 0px 0px 16px 0px rgba(0, 0, 0, 0.18);
  -moz-box-shadow: 0px 0px 16px 0px rgba(0, 0, 0, 0.18);
  box-shadow: 0px 0px 16px 0px rgba(0, 0, 0, 0.18);

  @include breakpoint(tablet-portrait-up) {
    margin: 0 1em;
    width: calc(100% - 2em);
    -webkit-box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.18);
    -moz-box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.18);
    box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.18);
  }

  &__highlight {
    width: 100%;
    height: 100%;
    position: absolute;
    right: 0;
    top: 0;
    clip-path: polygon(40% 0%, 100% 0%, 100% 100%, 100% 100%);
    opacity: 0.2;
  }

  &__header {
    width: 100%;
    display: flex;
    justify-content: center;
    position: relative;
    z-index: var(--z-index--level-1);

    .item-title {
      font-family: JackportCollege;
      letter-spacing: 0.2em;
      font-size: 20px;
      font-weight: 100;
    }
  }

  &__content {
    width: 100%;
    display: flex;
    flex-direction: row;
    position: relative;
    z-index: var(--z-index--level-1);

    .logo {
      width: 20%;
      display: flex;
      background: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAAAoCAYAAACM/rhtAAAABmJLR0QA/wD/AP+gvaeTAAAD0ElEQVRYhe2XT0ybZRjAf98X+tECbVnh6ygtk0JLgZmIWVRcnMHLMCExcDUhctxOO+2PiUcTx3Ve5skNEy9qXJZI7C5EnTpmTIgHKGtLdeVvK9/WNmtpC/28wOhS6PoWt2UJv1ub93mfX9/3+Z4+HxxyyCFlkUQDTo//Nizp0vfVJNMlfeTWxZM3RGKEBD8Yv23GYIicHfQ1dTstQnKh1RSf/zD/L5u5jpsX30lVGicLZZGV8b72JpOoHICnxUyf21Yv1yjjInElJzh4+fdfgZNFX932X3r71OnP7pwwKfIvn374mqnBZBAWBHiU3eSTr2fS6Y3Cu7c+7v+zYsGhy9NDsoGJXL5gK7d47L0Ovd+nCtdtMXfm4/q1qYWyeygGWctv6aM/XuifrAGQDUycG+q2dbaYn7b/geQA+n2q1O9Ty64Jr6ZsVyYDE0CzDJDLFyqRe250tpjJ5gpNADUH3WxuMcHdkMa95RTJdBYAa30tXS1m3vDa6HFZD7R/1YKxhxt8ObVAIrOFq0Wl7/hR6oy1AKQ3ssTXk3z1830a62oYG3BjbzQ+Ee8P6SV7DnpKK6gqweBKki/8ITxuJ32t9pLKtNSbsNSb6DhmJ7IYY/zGLGcGPXgd4u1JrA8Caw8zXPWH6Ov14HaWyhUjIdHhOsrrvZ1c9YeIJTaEBYVP8NpUBK/biWrbfajaLBLH7dCgSHw7WyiJUW0WPK+0cn0qwvnhHmDv69wLoROcW0yQyGzR7txtEyccEqeOSQTX4eZ8qdwO7jY72qM8gaWESEoxwemgRptDRdq+V5dFwtck8V2gQPiBTm5r/1gJCZdD5W5Qe3aCwZUU6pHdQn/VDtNLOtnNyuLtNgv3ViqeE8QFU5kcJqPy+LNZkYgmS9vFftSZakmmcyIpq2gzRbX9zR4PRFkq/y2PETpBs0khkxE7gWLSG1msdcrTFxYhJNjVaiamiT2FxcTWk8LNWkjwLY+N6EocvYq70tGJrsZ501t2oitBSLDbZeVIvYHIYkwoCcBCNEZzgwHRaVz4r25swE3o72XiWrLimJiWJPzPMh8NuEXTiQvaG42cGfQwMxtmIbpW9rp1XSccXWNmNszZ972oVuO+a/ejqmnG67BwYaSX61MRflqO4XKo2G2WJ8ctLcn9lTjNDQYujfRWJVe1IIDdauT8cA+BpQR/BDX+mouT2G7C1jqFrlYzYwPtwjX3vwnu0O200u082NRcDuEafN68HIKKQdbCq2JTxrMktJqiVpHXYbsG81v66JXJwMTOq96LplaR1/O5wuiL9jjkkJeC/wAuYDU2qFq2swAAAABJRU5ErkJggg==')
        no-repeat;

      background-position: center;
      background-size: 100%;

      &-image {
        object-fit: contain;
        width: 100%;
        align-self: center;
        box-sizing: border-box;
        background: var(--color--secondary);
      }
    }

    .detail {
      min-width: 60%;
      padding: 10px;

      &__label {
        text-transform: capitalize;
      }
    }
  }

  &__footer {
    display: flex;
    justify-content: flex-end;
    width: 100%;
    position: relative;
    z-index: var(--z-index--level-1);
  }
}
