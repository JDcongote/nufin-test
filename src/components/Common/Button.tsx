import React from 'react';
import './Button.scss';

type ButtonProps = {
  onClick: (data: any) => void;
  text?: string;
  floating?: string;
  context: any;
  visible?: boolean;
};

const Button = ({ text, onClick, context, floating, visible }: ButtonProps) => {
  function handleCLick(event: React.MouseEvent) {
    onClick(context);
  }
  return (
    <button
      className={`button button-primary ${floating}`}
      onClick={handleCLick}
      style={visible ? { display: 'block' } : { display: 'none' }}
    >
      {text}
    </button>
  );
};
Button.defaultProps = {
  visible: true
};
export default Button;
