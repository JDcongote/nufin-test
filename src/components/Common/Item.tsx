import React from 'react';
import './Item.scss';
import { Team } from 'redux-store/_types';

type ItemProps = {
  select?: (object: any) => void;
  items: ItemData[];
  title: string;
  image?: string;
  object?: any;
};

export type ItemData = {
  id: string;
  name: string;
};

const Item = ({ title, items, select, image, object }: ItemProps) => {
  function handleClick(event: React.MouseEvent<HTMLButtonElement>) {
    event.preventDefault();
    select(object);
  }
  return (
    <div className="team-item">
      <div className="team-item__header">
        <h3 className="school-name">{title}</h3>
      </div>
      <div className="team-item__content">
        <aside className="logo">
          <img className="logo-image" src={image}></img>
        </aside>
        <table className="detail">
          <tbody>
            {items.map(item => (
              <tr>
                <td className="" id={item.id + '-label'}>
                  {item.id}:
                </td>
                <td aria-labelledby={item.id + '-label'}>{item.name}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
      <div className="team-item__footer">
        <button onClick={handleClick}>Details</button>
      </div>
    </div>
  );
};

export default Item;
