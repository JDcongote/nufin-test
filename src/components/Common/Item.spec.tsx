import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';
import Item from './Item';

configure({ adapter: new Adapter() });

function createButton() {
  const onClick = jest.fn();
  const text = 'TEST_BUTTON';
  const context = { title: 'TEST' };
  return <Button context={context} onClick={onClick} text={text} />;
}

describe('Item', () => {
  it('Basic Item Test', () => {
    const items = [{ id: '1', name: 'a' }, { id: '2', name: 'b' }];
    const title = 'TITLE';
    const image = 'http://a.espncdn.com/i/teamlogos/ncaa/500-dark/2000.png';
    const button = createButton();
    const wrapper = mount(
      <Item items={items} title={title} image={image} button={button} />
    );

    const render = renderer.create(
      <Item items={items} title={title} image={image} button={button} />
    );

    const nodes = wrapper.find('.item-title');
    expect(nodes.text()).toBe(title);
    expect(wrapper.find('.logo-image').prop('src')).toBe(image);
    expect(render.toJSON()).toMatchSnapshot();
  });
});
