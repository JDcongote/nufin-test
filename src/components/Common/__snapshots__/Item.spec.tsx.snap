// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Basic Item Test 1`] = `
<div
  className="item"
>
  <div
    className="item__highlight"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />
  <div
    className="item__header"
  >
    <h3
      className="item-title"
    >
      TITLE
    </h3>
  </div>
  <div
    className="item__content"
  >
    <aside
      className="logo"
    >
      <img
        className="logo-image"
        src="http://a.espncdn.com/i/teamlogos/ncaa/500-dark/2000.png"
      />
    </aside>
    <table
      className="detail"
    >
      <tbody>
        <tr>
          <td
            className="detail__label"
            id="1-label"
          >
            1
            :
          </td>
          <td
            aria-labelledby="1-label"
          >
            a
          </td>
        </tr>
        <tr>
          <td
            className="detail__label"
            id="2-label"
          >
            2
            :
          </td>
          <td
            aria-labelledby="2-label"
          >
            b
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="item__footer"
  >
    <button
      className="button button-primary undefined"
      onClick={[Function]}
      style={
        Object {
          "display": "block",
        }
      }
    >
      TEST_BUTTON
    </button>
  </div>
</div>
`;
